app.js

```diff
@@ -35,6 +35,44 @@ async function hashText(message) {
   return hash384;
 }
 
+function parseContentType(res) {
+  const type = res.headers.get("content-type").split(";");
+
+  const REGEX = /^text\/css$/;
+  const isStylesheet = REGEX.test(type[0]);
+
+  if (isStylesheet) {
+    return 'style';
+  }
+  return 'script';
+}
+
+async function integrityMetadata(text) {
+  const hashBuffer = await hashText(text); // Array Buffer
+  const base64string = btoa(
+    String.fromCharCode(...new Uint8Array(hashBuffer))
+  );
+
+  return `sha384-${base64string}`;
+}
+
+async function displayResult(resultDiv, url, contentType, text) {
+  const integrity = await integrityMetadata(text);
+  const scriptEl = `<span style="color: #ffa07a">&lt;script src=</span><span style="color:#abe338">&quot;${encodeURI(
```

Comment:
I think it's generally better to generate both result HTML strings within the if/else below. No need to create both and then just use one.